test/test_dicom_to_nifti.py

```diff
@@ -8,16 +8,35 @@
 from shimmingtoolbox import __dir_testing__
 
 
-def test_dicom_to_nifti():
+def test_dicom_to_nifti_unsorted():
     # TODO: put as decorator
     with tempfile.TemporaryDirectory(prefix='st_'+pathlib.Path(__file__).stem) as tmp:
         path_nifti = os.path.join(tmp, 'nifti')
         subject_id = 'sub-test'
-        dicom_to_nifti(__dir_testing__, path_nifti, subject_id=subject_id)
+        # Conversion of the B0 mapping dicoms
+        dicom_to_nifti(os.path.join(__dir_testing__, 'dicom_unsorted'), path_nifti, subject_id=subject_id)
+
         # Check that all the files (.nii.gz and .json) are created with the expected names. The test data has 6
         # magnitude and phase data.
         for i in range(1, 7):
             for modality in ['phase', 'magnitude']:
                 for ext in ['nii.gz', 'json']:
```

Comment:
It is not really a good way to have more than two nested for loops. You should use itertools that helps to have only one for loop:
```suggestion
import itertools

        # Check that all the files (.nii.gz and .json) are created with the expected names. The test data has 6
        # magnitude and phase data.
        for i, modality, ext in itertools.product(range(1,7), ['pahse','magnitude'], ['nii.gz','json']:
```